src/Init/Data/List/Lemmas.lean

```diff
@@ -33,6 +33,17 @@ For each `List` operation, we would like theorems describing the following, when
 
 Of course for any individual operation, not all of these will be relevant or helpful, so some judgement is required.
 
+General principles for `simp` normal forms for `List` operations:
+* Arithmetic operations are "light", so e.g. we prefer to simplify `(L.drop i).drop j` to `L.drop (i + j)`,
+  rather than the other way round.
+* Function compositions are "light", so we prefer to simplify `(L.map f).map g` to `L.map (g ∘ f)`.
+* We generally prefer to move "more basic operations inwards", so for example
+  `List.map_reverse : (L.reverse.map f) = (L.map f).reverse` is a `simp` lemma,
+  because `List.map` is a more basic operation than `reverse`
+  (and there is more chance that we "know how to calculate `L.map f`")
+* We try to avoid non-linear left hand sides (i.e. with subexpressions appearing multiple times),
+  but this is only a weak preference.
+
 -/
```

Comment:
Maybe a bullet point that conversion functions (e.g. `toArray`) move inwards most aggressively, so that if an expression can be converted completely, it should so?